### ISSUE TITLE
Dock item alerting broken for tabbed items

### ISSUE THREAD

**Issue by e-calder** [nucleic/enaml#395]
Hi, 

Dock item alerting doesn't work for dock items that are tabbed inside a TabLayout:

**Repro:**
1. Run the dock_item_alerts example `enaml-run dock_item_alerts.enaml`

2. Drag the information dock item to the area with the Dummy items, so it ends up tabbed.

![EnamlDrag](https://user-images.githubusercontent.com/37456137/73546104-88109e00-4434-11ea-9d23-c973302742b8.PNG)

3. Now click the "Information" push button to alert it. You should see the error below:

```
Traceback (most recent call last):
  File "c:\users\[...]\enaml\qt\docking\q_dock_tab_widget.py", line 163, in _onAlerted
    self._snapAlertPixmaps(index, level)
  File "c:\users\[...]\enaml\qt\docking\q_dock_tab_widget.py", line 194, in _snapAlertPixmaps
    painter.initFrom(self)
AttributeError: 'QStylePainter' object has no attribute 'initFrom'
```

**Comment by MatthieuDartiailh** [comment]
Thanks for the report. I will try to have a look at the issue as soon as possible.

**Comment by MatthieuDartiailh** [comment]
It appears that replacing `painter` by `opt` avoid the crash. However I see weird glitches where the text is drawn too large or too big depending where I place the call to `initFrom`. Since I am on MacOS currently I am wondering if it is related to some HighDPI issue. Could you check on your setup if the display behaves as expected ?

**Comment by e-calder** [comment]
Hey

Thanks for the speedy response!

If I replace `painter` with `opt`, I agree that the text doesn't look right at all during the flash state:

![glitch](https://user-images.githubusercontent.com/37456137/73595281-bb276000-450e-11ea-82c4-fa6bd2aebbd4.gif)


If I totally remove both the `painter.initFrom(self)` lines, it's less glitchy but still not _quite_ right:

![glitch2](https://user-images.githubusercontent.com/37456137/73595290-deeaa600-450e-11ea-8505-f7605f6b50b2.gif)

I will have some more time to play about with this on Monday if we haven't figured out an optimal solution by then.

Thanks

**Comment by sccolbert** [comment]
I know this worked when I wrote it. It may be an incompatibility in either
a new version of PyQt or Pyside (depending on which you're using).

**Comment by e-calder** [comment]
Hey,

Indeed, it looks like `initFrom` existed in Qt 4.8:

https://doc.qt.io/archives/qt-4.8/qstylepainter-members.html

but not anymore in Qt 5.14:

https://doc.qt.io/qt-5/qstylepainter-members.html

**Comment by MatthieuDartiailh** [comment]
You can have a look at the PR. It seems to fix the issue locally but only copy the font. Still need to figure out the pen and background.

**Comment by MatthieuDartiailh** [comment]
Fixed in #396

**Comment by e-calder** [comment]
Thanks Matthieu, much appreciated
